base_geoengine/static/src/js/views/geoengine_view.js

```diff
@@ -229,39 +232,62 @@ var GeoengineView = View.extend(geoengine_common.GeoengineMixin, {
      * cfg - {Object} config object specific to the vector layer
      * data - {Array(features)}
      */
-    createVectorLayer: function(cfg, data) {
+
+    addFeatureToSource: function(cfg, data, vectorSource) {
         var self = this;
-        var geostat = null;
-        var vectorSource = new ol.source.Vector({});
-        if (data.length === 0) {
-            return new ol.layer.Vector({
-                source: vectorSource,
-                title: cfg.name,
-            });
-        }
         _.each(data, function(item) {
             var attributes = _.clone(item);
-            _.each(_.keys(self.geometry_columns), function(item) {
-                delete attributes[item];
-            });
+            _.each(_.keys(self.geometry_columns), function(col) {delete attributes[col]});
 
-            if (cfg.display_polygon_labels === true) {
-                attributes.label = item[cfg.attribute_field_id[1]];
-            }
-            else {
-                attributes.label = '';
-            }
+            attributes.label = (!!cfg.display_polygon_labels) ? item[cfg.attribute_field_id[1]] : '';
+            attributes.model = (!!cfg.model_id) ? cfg.model_id[2] : self.model;
 
             var json_geometry = item[cfg.geo_field_id[1]];
+
             if (json_geometry) {
-                vectorSource.addFeature(
-                    new ol.Feature({
-                        geometry: new ol.format.GeoJSON().readGeometry(json_geometry),
-                        attributes: attributes,
-                    })
-                );
+              vectorSource.addFeature(
+                  new ol.Feature({
+                      geometry: new ol.format.GeoJSON().readGeometry(json_geometry),
+                      attributes: attributes,
+                  })
+              );
+          }
+        })
+
+    },
+
+    createVectorLayer: function(cfg, data) {
+        var self = this;
+        if (!!cfg.model_id){
+
+            var model = new Model(cfg.model_id[2]);
+            var fields_to_read = [cfg.geo_field_id[1]];
```

Comment:
@lmarion-source In odoo, the field display_name is present by default on all the models. You should add this field by defaults into the list of retrieved fields and use this field as title of the details form dialog..

```js
var fields_to_read = [cfg.geo_field_id[1], 'display_name'];
```